File: TransCoder_tokenized_test_set_functions/python/PRINT_SHORTEST_PATH_PRINT_STRING_SCREEN.py
def printPath ( str ) :
    i = 0
    curX = 0
    curY = 0
    while ( i < len ( str ) ) :
        nextX = int ( ( ord ( str [ i ] ) - ord ( 'A' ) ) / 5 )
        nextY = ( ord ( str [ i ] ) - ord ( 'B' ) + 1 ) % 5
        while ( curX > nextX ) :
            print ( "Move Up" )
            curX -= 1
        while ( curY > nextY ) :
            print ( "Move Left" )
            curY -= 1
        while ( curX < nextX ) :
            print ( "Move Down" )
            curX += 1
        while ( curY < nextY ) :
            print ( "Move Right" )
            curY += 1
        print ( "Press OK" )
        i += 1
    

File: TransCoder_tokenized_test_set_functions/java/PRINT_SHORTEST_PATH_PRINT_STRING_SCREEN.java
static void printPath ( String str ) {
  int i = 0;
  int curX = 0, curY = 0;
  while ( i < str . length ( ) ) {
    int nextX = ( str . charAt ( i ) - 'A' ) / 5;
    int nextY = ( str . charAt ( i ) - 'B' + 1 ) % 5;
    while ( curX > nextX ) {
      System . out . println ( "Move Up" );
      curX --;
    }
    while ( curY > nextY ) {
      System . out . println ( "Move Left" );
      curY --;
    }
    while ( curX < nextX ) {
      System . out . println ( "Move Down" );
      curX ++;
    }
    while ( curY < nextY ) {
      System . out . println ( "Move Right" );
      curY ++;
    }
    System . out . println ( "Press OK" );
    i ++;
  }
}